data/js/about-home.js

```diff
@@ -8,27 +8,25 @@ var searchIconAndTextContainer = document.querySelector('#searchIconAndTextConta
  * @param {string} snippetContent - The content to inject
  */
 function showSnippet(snippetContent) {
-    var button;
     var addonSnippet = document.querySelector('#allaboard_snippet');
+    var button;
 
     // if a previous snippet added by the addon exists, remove it.
     if (addonSnippet) {
         addonSnippet.remove();
     }
 
     searchIconAndTextContainer.insertAdjacentHTML('afterend', snippetContent);
-
     button = document.querySelector('#allaboard_snippet_intent');
 
-    // not all snippets have buttons, so test before adding the event listener
-    if (button) {
-        button.addEventListener('click', function() {
-            if (typeof button.dataset !== 'undefined') {
-                // pass the button intent to the add-on
-                self.port.emit('intent', button.dataset['intent']);
-            }
+    button.addEventListener('click', function() {
```

Comment:
I definitely want to switch everything over to ES6 where it makes sense but, for right now, I reckon the main focus is to get the new features and related changes complete in order to move to QA starting tomorrow.